squarelet/email_api/serializers.py

```diff
@@ -0,0 +1,30 @@
+# Third Party
+from allauth.account.models import EmailAddress
+from rest_framework import serializers
+
+
+class PressPassEmailAddressSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = EmailAddress
+        fields = ("email", "verified", "primary")
+
+    def validate(self, attrs):
+        request = self.context.get("request")
+        if request.data.get("action") == "update":
```

Comment:
I know you mentioned this, but yeah - definitely do not do this.

The serialize is meant to serialize and de-serialize between the API input (JSON or form encoded) and the Python model - validate should validate that the JSON is valid for the Python model.  On deletion, there is no data, just delete this instance, so the serializer should not be invoked.  I think the validation for deletion can just happen in the View and that will be fine.  Then validate can just focus on the update and create actions.